charts/kong/templates/service-kong-admin.yaml

```diff
@@ -33,3 +34,55 @@ spec:
   selector:
     {{- include "kong.selectorLabels" . | nindent 4 }}
 {{- end -}}
+{{- else -}} {{/* Modern admin handler */}}
+{{- if and .Values.admin.enabled (or .Values.admin.http.enabled .Values.admin.tls.enabled) -}}
```

Comment:
Let's move `if .Values.admin.enabled` out to the top.
This value is being checked in both the if else blocks.

Reply:
This is intentional--the idea is that these are otherwise complete admin Service templates within the `{{- if .Values.admin.containerPort -}}` cases, and that removing the legacy behavior will remove the first case (and legacy if block) entirely, leaving the contents of the second case alone.